File: web/src/main/java/fr/arsenelapostolet/professor/server/core/entities/Grade.java
package fr.arsenelapostolet.professor.server.core.entities;

import java.math.BigDecimal;

public class Grade {

    private final BigDecimal score;
    private final Deliverable deliverable;

    public Grade(BigDecimal score, Deliverable deliverable) {
        this.score = score;
        this.deliverable = deliverable;
    }

    public BigDecimal getScore() {
        return score;
    }

    public Deliverable getDeliverable() {
        return deliverable;
    }
}

File: app/src/main/kotlin/fr/arsenelapostolet/professor/core/entities/Grade.kt
package fr.arsenelapostolet.professor.core.entities

import java.math.BigDecimal

class Grade(var score: BigDecimal, val deliverable: String) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Grade

        if (score != other.score) return false
        if (deliverable != other.deliverable) return false

        return true
    }

    override fun hashCode(): Int {
        var result = score.hashCode()
        result = 31 * result + deliverable.hashCode()
        return result
    }
}
